### PR TITLE
feat(apigatewayv2): http api - mTLS support 

### LINKED ISSUE #12559: (apigatewayv2): Expose DomainName mTLS config
API Gateway V2 supports mTLS [[docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-mutual-tls.html)]. It is exposed on the API Gateway V1 [L2 construct](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.DomainName.html), but not on the [V2 L2 construct](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigatewayv2.DomainName.html).

### Use Case

Blocks usage of the L2 construct for API Gateway V2.

### Proposed Solution

Extend [this PR](https://github.com/aws/aws-cdk/commit/eb2c568e6b87c66422137c2ef147011205195e0b#diff-36349b38b4709b1b2946a119fa13b2952d4ccfe9a88d7a1180b428c7b51a88a8) to API Gateway V2.

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by serahisaactho**
Resolves #12559 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2/README.md
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/README.md:249]:
```suggestion
  },
```

File: packages/@aws-cdk/aws-apigatewayv2/README.md
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/README.md:246]:
```suggestion
import * as s3 from '@aws-cdk/aws-s3';
const certArn = 'arn:aws:acm:us-east-1:111111111111:certificate';
const domainName = 'example.com';
const bucket = new s3.Bucket.fromBucketName(stack, 'TrustStoreBucket', ...);

new DomainName(stack, 'DomainName', {
  domainName,
  certificate: Certificate.fromCertificateArn(stack, 'cert', certArn),
  mtls: {
    bucket,
```

**Review by nija-at — DISMISSED** [review]
Thanks for submitting this PR @serahisaactho.

**Comment by nija-at** [comment]
@serahisaactho - once approved, please do not update the branch from master. Our merge bot will do everything for you.

Any manual changes you make will revert the approval.

**Review by nija-at — APPROVED** [review]

**Comment by serahisaactho** [comment]
@nija-at Sorry! Thanks for taking a look

**Comment by serahisaactho** [comment]
@nija-at It looks like the build is not successful which is blocking the merge. Can I update my branch to see what is causing the build failure?

**Comment by nija-at** [comment]
That's weird that the build failed. I've restarted the build.

**Review by nija-at — DISMISSED** [review]

**Review by nija-at — APPROVED** [review]

**Review by nija-at — CHANGES_REQUESTED** [review]
@serahisaactho - Your unit tests are failing. See the build logs linked above.

**Comment by serahisaactho** [comment]
@nija-at Will fix them. Is there another command to run the unit tests besides `yarn build`? The unit tests still seem to be passing locally for me.

**Comment by serahisaactho** [comment]
@nija-at I've updated the unit tests. The build is succeeding now.

**Pull request merged**
